app/app.go

```diff
@@ -0,0 +1,405 @@
+package app
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmos "github.com/tendermint/tendermint/libs/os"
+	dbm "github.com/tendermint/tm-db"
+
+	bam "github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	distr "github.com/cosmos/cosmos-sdk/x/distribution"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	"github.com/cosmos/cosmos-sdk/x/slashing"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/supply"
+	mesgcodec "github.com/mesg-foundation/engine/codec"
+	"github.com/mesg-foundation/engine/cosmos"
+	executionsdk "github.com/mesg-foundation/engine/sdk/execution"
+	instancesdk "github.com/mesg-foundation/engine/sdk/instance"
+	ownershipsdk "github.com/mesg-foundation/engine/sdk/ownership"
+	processsdk "github.com/mesg-foundation/engine/sdk/process"
+	runnersdk "github.com/mesg-foundation/engine/sdk/runner"
+	servicesdk "github.com/mesg-foundation/engine/sdk/service"
+)
+
+const appName = "engine"
+
+var (
+	// DefaultCLIHome is the default home directories for the application CLI
+	DefaultCLIHome = os.ExpandEnv("$HOME/.mesg-cli")
+
+	// DefaultNodeHome sets the folder where the applcation data and configuration will be stored
+	DefaultNodeHome = os.ExpandEnv("$HOME/.mesg/tendermint")
+
+	// ModuleBasics The module BasicManager is in charge of setting up basic,
+	// non-dependant module elements, such as codec registration
+	// and genesis verification.
+	ModuleBasics = module.NewBasicManager(
+		genutil.AppModuleBasic{},
```

Comment:
the params module is not present here:
```
params.AppModuleBasic{},
```
but i'm not sure it's useful. let's check the template and simapp to know

Reply:
It is used to create subspaces as well, let's keep it for now.